hal/src/main/native/include/hal/DriverStation.h

```diff
@@ -66,6 +66,15 @@ int32_t HAL_SendConsoleLine(const char* line);
  */
 int32_t HAL_GetControlWord(HAL_ControlWord* controlWord);
 
+/**
+ * Gets the current operating mode of the driver station.
+ *
+ * @param buf buffer for operating mode string (out)
+ * @param len length of buffer (in), length of string (out)
+ * @return the error code, or 0 for success
+ */
+int32_t HAL_GetOpMode(char* buf, int32_t* len);
```

Comment:
Not the _biggest_ fan of this being buf vs length, when we have WPI_String. However, I'm also concerned about the allocations, since this is going to be on the hot path. Need to think about this one.

Reply:
Yeah I considered WPI_String, but this is called likely multiple times every loop.  Can’t really help Java though, that’s always going to be an allocation.

Reply:
The other approach would be to pass a *mutable* WPI_String, but as we don't do that basically anywhere else in the API, and is probably not going to be wrapper-friendly, that feels like a bad idea.